index.js

```diff
@@ -42,21 +56,52 @@ var
     return cWhere (collection, predicates);
   };
 
-var objectList = function objectList (collection) {
-  return {
-    getByKey: function (key) {
-      return getByKey.apply(null, [collection, key]);
-    },
-    whitelist: function (keyWhitelist) {
-      return whitelist.apply(null, [collection, keyWhitelist]);
-    },
-    concat: function () {
-      return concat(collection);
-    },
-    where: function (keyWhitelist) {
-      return where.apply(null, [collection, keyWhitelist]);
-    },
-  };
+objectList = function objectList (options) {
+  var
+    collection,
+    api = {
+      getByKey: function (key) {
+        return getByKey.apply(null, [collection, key]);
+      },
+      whitelist: function (keyWhitelist) {
+        return whitelist.apply(null, [collection, keyWhitelist]);
+      },
+      concat: function () {
+        return concat(collection);
+      },
+      where: function (keyWhitelist) {
+        return where.apply(null, [collection, keyWhitelist]);
+      },
+      add: function (records, cb) {
```

Comment:
`cb` is a param because then I don't need to add `arguments[1]` onto the array passed to `add.apply`, instead I just name it. The extra param will be ignored if it's not an async invocation. `cb` will be the node-ized deferred object from Q when `add` is invoked as async.

The `records` param is the new record(s) being added. Let me know if this is not how you want it implemented, I can change it.